Automattic/simplenote-electron
# Improve a11y for modals

These improvements currently apply to the Share, Settings, and About dialogs.

### A11y Improvements

- <kbd>Esc</kbd> will close the dialog.
- Close dialog buttons are clearly labeled with `aria-label`.
- Keyboard focus is correctly trapped inside the dialog.
- Dialog will be correctly `aria-label`ed even if a visible heading does not appear within the dialog (Like in the About dialog).

### Technical Changes

- The `multiple` (previously `single`) property of the dialog redux data now defaults to `false` so we don't have to set it every time.
- The `modal` property of the dialog redux data has been removed.
- Dialog settings (`type` and `title`) now live in a single place (`lib/dialogs/types`).

### To Test

In `scss/buttons.scss`, remove the `outline: 0` rules from `button` and `.button` so you can see the focus outlines. (FYI: These focus outlines will be addressed in a separate PR)

Some things to check:

- Dialogs look correct in both Light and Dark mode.
- <kbd>Esc</kbd> closes a dialog.
- Keyboard focus is trapped (Test by <kbd>tab</kbd>ing around immediately after the dialog has opened).
- A dialog element (the one with `role="dialog"`) has an `aria-label`. (Check in devtools)
